### PR TITLE
Set AGENT_TOOLSDIRECTORY in packer template

### PR DESCRIPTION
**Pull request by ryanspletzer**
# Description
New tool, Bug fixing, or Improvement?  
Please include a summary of the change and which issue is fixed. Also include relevant motivation and context.  
**For new tools, please provide total size and installation time.**

This was set in the ruby installer but it is referenced elsewhere so I
believe it makes the most sense to have this set centrally in the packer
template to be referenced across multiple scripts.

#### Related issue:
#2456 

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/windows2016.json [hunk 1]
```diff
@@ -138,7 +139,8 @@
             "type": "powershell",
             "inline": [
                 "setx ImageVersion {{user `image_version` }} /m",
-                "setx ImageOS {{user `image_os` }} /m"
+                "setx ImageOS {{user `image_os` }} /m",
+                "setx AGENT_TOOLSDIRECTORY {{user `agent_tools_directory` }} /m"
```
Comment (miketimofeev) [inline images/win/windows2016.json:143]: Maybe move all this inline-block to https://github.com/actions/virtual-environments/blob/main/images/win/scripts/Installers/Initialize-VM.ps1?

Comment (al-cheb) [reply images/win/windows2016.json:143]: Hi, @ryanspletzer 
Could you please merge the latest changes from main branch and  put  `setx AGENT_TOOLSDIRECTORY` near the line https://github.com/actions/virtual-environments/blob/1eb504f4d48c428bf25a4c0fce604e98d8ac6f4c/images/win/scripts/Installers/Initialize-VM.ps1#L36 ?

Comment (ryanspletzer) [reply images/win/windows2016.json]: Yep, done and done.

File: images/win/windows2016.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
         "imagedata_file": "C:\\imagedata.json",
         "helper_script_folder": "C:\\Program Files\\WindowsPowerShell\\Modules\\",
         "psmodules_root_folder": "C:\\Modules",
+        "agent_tools_directory": "C:\\hostedtoolcache\\windows",
```
Comment (maxim-lobanov) [inline images/win/windows2016.json:24]: You should also add `AGENT_TOOLSDIRECTORY` to environment variables like you have done in Win19 template

**Comment by maxim-lobanov** [comment]
@ryanspletzer , sorry for delay. Changes look good except minor comment in Win2016 template. Could you please fix it and pull latest main. After that we will be able to merge

File: images/win/windows2016.json (hunk 2 above)
Comment (ryanspletzer) [reply images/win/windows2016.json:24]: Ah yep good catch, merged with main and added that.

https://github.com/actions/virtual-environments/pull/2457/files/8abd479475a4b4be8da50d1e4c7fc0b8853bf18b..1195d292eec2f950a1f8fdf1c8db3e183364532f

**Comment by ryanspletzer** [comment]
@maxim-lobanov Updated. :)

**Review by maxim-lobanov — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Review by AlenaSviridenko — APPROVED** [review]

**Review by miketimofeev — APPROVED** [review]

**Comment by maxim-lobanov** [comment]
@ryanspletzer, merged. Thank you for submitting PR.

**Pull request merged**